crates/storage/src/traits/impls/mod.rs

```diff
@@ -12,172 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-macro_rules! impl_always_packed_layout {
-    ( $name:ident < $($frag:ident),+ >, deep: $deep:expr ) => {
-        impl<$($frag),+> $crate::traits::SpreadLayout for $name < $($frag),+ >
-        where
-            $(
-                $frag: $crate::traits::PackedLayout,
-            )+
-        {
-            const FOOTPRINT: ::core::primitive::u64 = 1_u64;
-            const REQUIRES_DEEP_CLEAN_UP: ::core::primitive::bool = $deep;
-
-            #[inline]
-            fn pull_spread(ptr: &mut $crate::traits::KeyPtr) -> Self {
-                $crate::traits::impls::forward_pull_packed::<Self>(ptr)
-            }
-
-            #[inline]
-            fn push_spread(&self, ptr: &mut $crate::traits::KeyPtr) {
-                $crate::traits::impls::forward_push_packed::<Self>(self, ptr)
-            }
-
-            #[inline]
-            fn clear_spread(&self, ptr: &mut $crate::traits::KeyPtr) {
-                $crate::traits::impls::forward_clear_packed::<Self>(self, ptr)
-            }
-        }
-
-        impl<$($frag),+> $crate::traits::SpreadAllocate for $name < $($frag),+ >
-        where
-            Self: ::core::default::Default,
-            $(
-                $frag: $crate::traits::PackedAllocate,
-            )+
-        {
-            #[inline]
-            fn allocate_spread(ptr: &mut $crate::traits::KeyPtr) -> Self {
-                $crate::traits::impls::forward_allocate_packed::<Self>(ptr)
-            }
-        }
-    };
-    ( $name:ty, deep: $deep:expr ) => {
-        impl $crate::traits::SpreadLayout for $name
-        where
-            Self: $crate::traits::PackedLayout,
-        {
-            const FOOTPRINT: ::core::primitive::u64 = 1_u64;
-            const REQUIRES_DEEP_CLEAN_UP: ::core::primitive::bool = $deep;
-
-            #[inline]
-            fn pull_spread(ptr: &mut $crate::traits::KeyPtr) -> Self {
-                $crate::traits::impls::forward_pull_packed::<Self>(ptr)
-            }
-
-            #[inline]
-            fn push_spread(&self, ptr: &mut $crate::traits::KeyPtr) {
-                $crate::traits::impls::forward_push_packed::<Self>(self, ptr)
-            }
-
-            #[inline]
-            fn clear_spread(&self, ptr: &mut $crate::traits::KeyPtr) {
-                $crate::traits::impls::forward_clear_packed::<Self>(self, ptr)
-            }
-        }
-
-        impl $crate::traits::SpreadAllocate for $name
-        where
-            Self: $crate::traits::PackedLayout + ::core::default::Default,
-        {
-            #[inline]
-            fn allocate_spread(ptr: &mut $crate::traits::KeyPtr) -> Self {
-                $crate::traits::impls::forward_allocate_packed::<Self>(ptr)
-            }
-        }
-    };
-}
-
 mod arrays;
-mod collections;
 mod prims;
 mod tuples;
```

Comment:
So these three modules just include tests and no real code related to trait implementations. I'd either move them here, or to to some tests module